common/djangoapps/course_modes/views.py

```diff
@@ -103,12 +102,7 @@ def get(self, request, course_id, error=None):  # lint-amnesty, pylint: disable=
 
         # Check whether the user has access to this course
         # based on country access rules.
-        embargo_redirect = embargo_api.redirect_if_blocked(
-            course_key,
-            user=request.user,
-            ip_address=get_client_ip(request)[0],
-            url=request.path
-        )
+        embargo_redirect = embargo_api.redirect_if_blocked(request, course_key)
```

Comment:
I changed this "high level" API to just take a request and go from there (felt like a better boundary than requiring all callers to know how to get the right IP address from the request, specifically since that has gotten more complicated).